Add documentation for the new watson_iot component

mtreinish opened the pull request:
**Description:**

This commit adds documention for the new watson_iot component which was
added in home-assistant/home-assistant#13664
It links to the official documentation for setting up the cloud service
and generating the required auth information.


**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#13664

## Checklist:

- [X] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [X] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

mtreinish committed:
Add documentation for the new watson_iot component

This commit adds documention for the new watson_iot component which was
added in home-assistant/home-assistant#13664
It links to the official documentation for setting up the cloud service
and generating the required auth information.

frenck committed:
:pencil2: Missing comma

frenck commented on source/_components/watson_iot.markdown:
````diff
@@ -0,0 +1,70 @@
+  token: 'auth_token'
+```
+
+Configuration variables:
````
Please use the configuration tags, for more information please see:
https://home-assistant.io/developers/documentation/create_page/#configuration

mtreinish replied:
Ok, I pushed a patch to use the configuration tags

frenck commented:
Thank you for this PR @mtreinish! I've reviewed it and found one abnormality.
Could you please take a look? 👍

mtreinish committed:
Use configuration tags

This commit updates the formatting of the configuration section of the
documentation. Previously it manually formatted a list instead of
leveraging the templating format. This corrects that oversight and uses
the standard format.

frenck commented:
Thanks for updating the PR @mtreinish! 👍 
✅ Approved, can be merged as soon as the parent PR gets merged.

frenck committed:
:art: Adds logo

frenck committed:
:arrow_up: ha_release -> 0.68

arsaboo commented on source/_components/watson_iot.markdown:
```diff
@@ -0,0 +1,104 @@
+
+To use this component, you first need to register a gateway device type and then
+a gateway device in your IoT platform instance. For instructions on how to do
+this check the [official documentation](https://console.bluemix.net/docs/services/IoT/gateways/dashboard.html#IoT_connectGateway)
```
Please indicate whether it is FREE or paid so that users are not wasting their time.

frenck replied:
They have a free tier, which is sufficient imho.

arsaboo commented on source/_components/watson_iot.markdown:
```diff
@@ -0,0 +1,104 @@
+footer: true
+logo: ibm.png
+ha_category: History
+ha_release: 0.68
```
Remember to update the release.

arsaboo commented on source/_components/watson_iot.markdown:
```diff
@@ -0,0 +1,104 @@
+ha_release: 0.68
+---
+
+The `watson_iot` component enables you to link the devices in home-assistant
```
`home-assistant` -> `Home Assistant`

frenck committed:
:pencil2: Processed RFC's

boojew commented on source/_components/watson_iot.markdown:
```diff
@@ -0,0 +1,104 @@
+
+To use this component, you first need to register a gateway device type and then
+a gateway device in your IoT platform instance. For instructions on how to do
+this check the [official documentation](https://console.bluemix.net/docs/services/IoT/gateways/dashboard.html#IoT_connectGateway)
```
Could I suggest adding a line about why you would use this component? "i.e. you can use this to bring IoT history data into Watson to build an application..." My thought is that people will see Watson and think they will get some out of the box analytics or machine learning.
